ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/rest/RestTestExecutor.java

```diff
@@ -759,7 +777,10 @@ private static void matchResponseFields(
             (Map<String, Object>) expectedValue,
             statements,
             idx,
-            newPath
+            newPath,
+            actualRecords,
+            expectedRecords,
+            pathIndex
         );
       } else {
         if (operator == MatchOperator.STARTS_WITH) {
```

Comment:
Do we want to add some operator to add to the query name or something like that, which will enable this behavior on a per test basis or do we just always want it to behave this way? I think we probably care about ordering in some cases, like push queries where at least in some scenarios there a well defined ordering.  The same is true of pull queries, and especially will be when we add a ORDER BY clause.  Or it could be inverted to be order agnostic unless specified.

Reply:
Good point. 

I guess if as a next step we want to [parallelize partition fetching ](https://github.com/confluentinc/ksql/pull/8161) we would need to enable it by default (otherwise pretty much all current functional tests that don't have partitions=1 will fail) but add an option to disable it in case ordering is needed. Can you explain what you meant by adding an operator? Alternatively do you think it would be possible to control this via some internal property that can be set in the functional test in the property section? e.g.,

```
      "properties": {
        "ksql.rqtt.ordering.enforce":  true
      }
```

Reply:
Added OutputConditions to control whether or not ordering is enforced, thanks for the suggestion @AlanConfluent . Default is ordering is not enforced.